app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarkerInformationPopover.tsx

```diff
@@ -0,0 +1,440 @@
+import * as React from 'react';
+
+import { Application } from 'core/application/application.model';
+import { IExecution, IPipeline, IExecutionStageSummary } from 'core/domain';
+import { relativeTime, timestamp, duration } from 'core/utils';
+import { Spinner } from 'core';
+import { PipelineGraph } from '../../config/graph/PipelineGraph';
+import { AccountTag } from 'core/account';
+
+import { ExecutionInformationPopoverState } from './ExecutionInformationPopoverState';
+import './executionMarkerInformationPopover.less';
+
+export interface IExecutionErrorLocatorProps {
+  application: Application;
+  target: Element;
+  execution: IExecution;
+  stage: IExecutionStageSummary;
+  onClose: Function;
+}
+
+export interface IExecutionLocatorState {
+  pipelineConfigDetails: IPipeline;
+  executionDetails: IExecution;
+  failedInApplication: string;
+  link: string;
+  showPipelineGraph: boolean;
+  stageDetails: any;
+}
+
+export class ExecutionMarkerInformationPopover extends React.PureComponent<
+  IExecutionErrorLocatorProps,
+  IExecutionLocatorState
+> {
+  private allExecutions: any[];
+  private arrowLeftOffset: number;
+  private childExecution: any;
+  private childPipelineConfig: any;
+  private childTerminalPipelineStage: any;
+  private informationIconHeight: number;
+  private informationPopoverContainerScrollTop: number;
+  private informationState: ExecutionInformationPopoverState;
+  private popoverHalfWidth: number;
+
+  constructor(props: IExecutionErrorLocatorProps) {
+    super(props);
+    this.state = {
+      executionDetails: null,
+      failedInApplication: null,
+      link: null,
+      pipelineConfigDetails: null,
+      showPipelineGraph: false,
+      stageDetails: null,
+    };
+
+    const target = document.querySelector('.all-execution-groups');
+
+    if (target) {
+      target.removeEventListener('scroll', this.onScroll);
+      target.addEventListener('scroll', this.onScroll);
+    }
+    window.removeEventListener('resize', this.onResize);
+    window.addEventListener('resize', this.onResize);
+
+    this.allExecutions = [];
+    this.arrowLeftOffset = 404;
```

Comment:
Again, these constants seem very fragile, and I'm worried about the long-term maintenance of this component. Do we feel that the benefits from having the modal adjust according to scroll and window resize justify this added complexity?

Reply:
i mentioned above bootstrap sets the popover to max 400 and is also controlled by the object it is associated with like the stage <TooltipComponent>{stageDetails}</TooltipComponent> because of this already being a tooltip and not using jquery, bootstrap to actually control this popover i had to define some values in the file.